toolkit/scripts/srpm_pack.mk

```diff
@@ -84,7 +83,11 @@ $(STATUS_FLAGS_DIR)/build_srpms.flag: $(local_specs) $(local_spec_dirs) $(local_
 $(STATUS_FLAGS_DIR)/build_toolchain_srpms.flag: $(STATUS_FLAGS_DIR)/build_srpms.flag
 	@touch $@
 else
-$(STATUS_FLAGS_DIR)/build_srpms.flag: $(chroot_worker) $(local_specs) $(local_spec_dirs) $(SPECS_DIR) $(go-srpmpacker) $(srpm_pack_list_file) $(local_spec_sources)
+
+# Dependencies common to both the full packer and the toolchain-only packer targets
+common_srpm_packer_deps = $(local_specs) $(local_spec_dirs) $(SPECS_DIR) $(local_spec_sources) $(go-srpmpacker) $(built_srpms) $(BUILD_SRPMS_DIR)
```

Comment:
We are abusing make a bit here. If we were doing make "correctly" we would have a rule for each package, and track each packages' dependencies individually in the makefile (probably generated dynamically using a utility script). But, instead we operate on the level of a single folder. If a .spec file, or a source hash, etc. are changed we want to regenerate the .src.rpm associated with it. Unfortunately since we don't have individual rules for each file we trigger the packer tool on everything.

This is "ok" though, since the packer is smart and will look at the timestamps of each input and skip ones that are still ok. This PR attempts to help Make understand that, and allows it to 'noop' if packer decides everything is actually fine still.

Agreed that maybe we could do something to be more selective with the files we watch based on the pack list. I'm not sure its worth the complexity though, I feel like we very rarely modify package files we aren't going to build right away, and the cost of running packer is quite light generally (a few seconds if it decides to noop).